CommonFunctions/Apply-Values.ps1

```diff
@@ -22,8 +22,18 @@ function Apply-Values($config, $values) {
 function Apply-StringValue($thisValue, $values) {
     [regex]::matches($thisValue, "{{.*?}}") | select -expand value |%{
         $origKey = $_
-        $valueKey = $_.Trim("{{}}")
-        $thisValue = $values["$valueKey"] | %{ $thisValue -replace $origKey, $_ }
+        $thisValue = Extract-Value-For $_ $values | %{ $thisValue -replace $origKey, $_ }
```

Comment:
Not Powershell convention to name functions like `Extract-Value-For`, better use `Extract-ValueFor`?